server/jetstream_api.go

```diff
@@ -1825,17 +1829,37 @@ func (s *Server) jsStreamInfoRequest(sub *subscription, c *client, a *Account, s
 	// Check if they have asked for subject details.
 	if subjects != _EMPTY_ {
 		if mss := mset.store.SubjectsState(subjects); len(mss) > 0 {
-			if len(mss) > JSMaxSubjectDetails {
-				resp.StreamInfo = nil
-				resp.Error = NewJSStreamInfoMaxSubjectsError()
-				s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
-				return
+			// As go iterates over map in a non-consistent order, no choice but to buffer it a slice
+			type SubjectsState struct {
+				subject  string
+				messages uint64
 			}
-			sd := make(map[string]uint64, len(mss))
+
+			buffer := make([]SubjectsState, 0, len(mss))
 			for subj, ss := range mss {
-				sd[subj] = ss.Msgs
+				buffer = append(buffer, SubjectsState{subj, ss.Msgs})
 			}
+
+			// Sort it
+			sort.Slice(buffer[:], func(i, j int) bool {
+				return buffer[i].subject < buffer[j].subject
+			})
+
+			limitSize := JSMaxSubjectDetails
+			sd := make(map[string]uint64, limitSize)
+			end := offset + limitSize
+			if end > len(buffer) {
+				end = len(buffer)
+			}
+
+			for _, ss := range buffer[offset:end] {
```

Comment:
You need to check for `offset`, not simply cap `end` because then you could have offset > end and get a panic "slice bounds out of range"

Reply:
Ah yes, good catch added protection against bad requests with offset out of range